src/ssl.h

```diff
@@ -41,6 +41,9 @@
  * this pragma selectively suppresses this type of warnings in clang
  */
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
This is not specific to OS X as far as I can see, rather specific to the Clang compiler - perhaps a specific version of it? Clang might be used instead of GCC on Linux as well.

Change `#ifdef __APPLE__` to `__clang__`.